### PR TITLE
feat: Add possibility to login using access_token

### PR DESCRIPTION
**Pull request by yuryfirebolt**

### CODE REVIEW

File: src/firebolt_cli/common_options.py [hunk 1]
```diff
@@ -63,6 +63,14 @@ def password_from_config_file(
         callback=default_from_config_file(DEFAULT_API_URL, required=False),
         hidden=True,
     ),
+    option(
+        "--access-token",
+        envvar="FIREBOLT_ACCESS_TOKEN",
+        help="Firebolt token for authentication, "
+        "if failed username/password will be used",
+        required=False,
+        hidden=True,
```
Comment (ptiurin) [inline src/firebolt_cli/common_options.py:72]: Any reason to make it hidden?

File: src/firebolt_cli/utils.py [hunk 2]
```diff
@@ -66,14 +68,24 @@ def construct_resource_manager(**raw_config_options: str) -> ResourceManager:
...
+    if raw_config_options["access_token"] is not None:
+        try:
+            return ResourceManager(
+                Settings(
+                    **settings_dict,
+                    access_token=raw_config_options["access_token"],
+                )
+            )
+        except HTTPStatusError:
+            pass
```
Comment (ptiurin) [inline src/firebolt_cli/utils.py:80]: lets add a test that tests this part as well.

File: src/firebolt_cli/common_options.py (hunk 1 above)
Comment (yuryfirebolt) [reply src/firebolt_cli/common_options.py:72]: Actually, this option is mostly needed, when using via Docker, but maybe it makes more sense to not make it hidden. I will change it

File: src/firebolt_cli/utils.py (hunk 2 above)
Comment (yuryfirebolt) [reply src/firebolt_cli/utils.py:80]: Done

**Review by ptiurin — APPROVED** [review]

**Pull request merged**